app/models/maintenance_tasks/run.rb

```diff
@@ -452,6 +454,15 @@ def task
 
     private
 
+    def instrument_status_change
+      return if status.nil? || !status_previously_changed?
+      return if status.in?(["running", "pausing", "cancelling", "interrupted"])
+
+      ActiveSupport::Notifications.instrument("maintenance_tasks.#{status}", run: self)
+    rescue
```

Comment:
I understand the idea to ensure that issuing a notification will not result in the whole flow being halted but I think this `rescue` overly defensive. Realistically we don't expect anything to go wrong in this method but `rescue` will swallow all the problems including even `NoMethodError`

Reply:
🤔 lol, I need another coffee.... this TaskJobTest passes if I don't add this `after_save`, but if I add this method it starts failing (and I'm fairly certain it's not this method making it fail, it's a validation error)

Reply:
Yeah removing the rescue clause is probably what makes the most sense. If we were to keep it, we should find a way to re-raise the exception from the job after the data is correctly persisted. Maybe we could add the rescue to the readme example to show the best practice that prevents the instrumentation block from raising.